src/Google.Cloud.Functions.Framework/FunctionException.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright 2020, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Net;
+
+namespace Google.Cloud.Functions.Framework
+{
+    // More decisions and considerations:
+    // - Another message to be set on the response? (As content? Something else?)
+    // - Prohibit success status codes?
```

Comment:
I'd say yes, although maybe provide another way to set different success status codes? But definitely wouldn't like an exception controlling different success statuses.

Reply:
It would certainly be an anti-pattern to do so. Whether that makes it worth explicitly prohibiting or not, I don't know... but I'm happy enough to do it.